xsd/src/schema_manager.js

```diff
@@ -237,7 +237,9 @@ SchemaManager.prototype.exportNamespaces = function() {
 	var self = this;
 	var namespacePrefixes = {};
 	var prefixUsed = {};
+	this.addNamespacePrefixes(namespacePrefixes, prefixUsed, {'xs': self.xsNS});
```

Comment:
Just for clarity: This is a purely "cosmetic" change that happens after all XSD parsing is finished. It affects only the prefixes chosen for the final "schema.namespace" sub structure and nothing else.

This has nothing to do with the bug fixes discussed here, and could have been proposed in a separate pull request.

The only reason I was including it here is because it ensures that most test cases remain stable after the fix. Otherwise, some namespace prefixes of the final result schema would change, leading to not-nice-looking namespace prefixes to the user.

Maybe this is easier demonstrated than explained: Just disable this and the other "addNamespacePrefixes" calls, and compare the generated schemas (for the then-failing test cases). That should increase the clarity.